crescent/ext/tasks/loop.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, Sequence
+
+from crescent.ext.tasks.task import TaskCallbackT, _link_task, _Task
+from crescent.internal import MetaStruct
+
+__all__: Sequence[str] = ("loop",)
+
+
+class _Loop(_Task):
+    def __init__(self, callback: TaskCallbackT, *, hours: int, minutes: int, seconds: int) -> None:
+        self.timedelta = hours * 3600 + minutes * 60 + seconds
+        self.first_loop = True
+
+        super().__init__(callback)
+
+    def time_to_next(self) -> float:
+        if self.first_loop:
+            self.first_loop = False
+            return 0
+
+        return self.timedelta
+
+
+def loop(
+    *, hours: int = 0, minutes: int = 0, seconds: int = 0
```

Comment:
Could be cool if this optionally accepted a timedelta object as a positional argument. Could use overloads to specify that you should only be using one or the other (other being kwargs)

Reply:
good idea. I think I prefer classmethods in this case though.

Reply:
nvm will use overloads to avoid a python builtins moment